### PR TITLE
cmd, consensus, eth, ethstats: modify consensus interface to handle different consensus messages

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: consensus/consensus.go [hunk 1]
```diff
@@ -92,6 +91,17 @@ type Engine interface {
 
 	// APIs returns the RPC APIs this consensus engine provides.
 	APIs(chain ChainReader) []rpc.API
+
+	// Protocol returns the protocol for this consensus
+	Protocol() Protocol
+}
+
+type Handler interface {
```
Comment (tailingchen) [inline consensus/consensus.go]: add comments for ``Handler

Comment (markya0616) [inline consensus/consensus.go]: OK

File: consensus/istanbul/backend/backend.go [hunk 2]
```diff
@@ -162,10 +191,10 @@ func (sb *backend) Commit(proposal istanbul.Proposal, seals [][]byte) error {
 	if _, err := sb.inserter(types.Blocks{block}); err != nil {
 		return err
 	}
-	msg := istanbul.NewCommittedEvent{
-		Block: block,
+
+	if sb.broadcaster != nil {
+		sb.broadcaster.BroadcastBlock(block, false)
```
Comment (tailingchen) [inline consensus/istanbul/backend/backend.go]: Use go routine to broadcast? No need to wait broadcasting here.

Comment (markya0616) [inline consensus/istanbul/backend/backend.go]: OK

File: consensus/consensus.go (hunk 1 above)
Comment (yutelin) [inline consensus/consensus.go]: is -> as?

File: eth/peer.go [hunk 3]
```diff
@@ -132,6 +132,12 @@ func (p *peer) MarkTransaction(hash common.Hash) {
 
 // SendTransactions sends transactions to the peer and includes the hashes
 // in its transaction hash set for future reference.
+func (p *peer) Send(msgcode uint64, data interface{}) error {
```
Comment (markya0616) [inline eth/peer.go]: OK

**Pull request merged**